src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1134,12 +1134,17 @@ - (void)_reportAttributes:(NSArray<NSDictionary<NSString *, id> *> *)attributes
     }
 }
 
-- (void)_handleAttributeReport:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport
+- (void)_handleAttributeReport:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport fromSubscription:(BOOL)isFromSubscription
 {
     std::lock_guard lock(_lock);
 
     // _getAttributesToReportWithReportedValues will log attribute paths reported
-    [self _reportAttributes:[self _getAttributesToReportWithReportedValues:attributeReport]];
+    [self _reportAttributes:[self _getAttributesToReportWithReportedValues:attributeReport fromSubscription:isFromSubscription]];
+}
+
+- (void)_handleAttributeReport:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport
```

Comment:
I think we should remove this overload, so that callers have to clearly decide whether they are "from subscription" or not.

For example, the unit test thing should be doing fromSubscription:YES, I think, or making that an argument or something.

Reply:
Agree - thought about it, but decided to go with the "smallest possible change" for the first pass.  I'll see how many call sites that change affects - might need some context from the team updating some of them.

Reply:
It affects 3 callsites: subscriptions, reads, and the unit test bit.

Reply:
I'm going to make a new PR for these changes in the coming hours.

Reply:
PR to remove convenience method: https://github.com/project-chip/connectedhomeip/pull/33489